Td xml parsing

dohnto opened the pull request:
Fixes #8, but...

I have no idea why @lksv switched from `text/template` to `html/template`, if there was some special reason.

Also, now one test fails:
```bash
# make test
--- FAIL: TestGenerateTemplate (0.00s)
	template_test.go:41: generateTemplate("K={{ .NONEXISTING }}", "test") => ("K=<no value>", _), want ("K=", _)
...
```

But 
```bash
# cat a 
K={{ .NONEXISTING }}
# goenvtemplator2 -template $PWD/a:/dev/stdout
K= 
```

Which seems correct, no idea why the test seems to return different value than the binary...

@lksv @freznicek @balous

FUSAKLA commented:
So.. i tested it on `go1.7.4, debian-jessie`

Sorry but I must ask..

Did you run also `make install`? Cause I tried it many times and the behavior is consistent in tests and in bash execution way. Just asking cause i see you run `goenvtemplator2 -template $PWD/a:/dev/stdout` without `./` and simple `make` runs only `build`. That`s the only way i managed to gain this inconsistency (`make build + make install` from `master` and than only `make` from your branche)

I know it's silly question but for the inconsistency it's the only thing i was able to reproduce.

---

Now the **real problem**. As i understand you are trying to return empty string if the key is missing.
Sadly that is not possible. At least with this `missingkey` option. 

According to the (documentation of missingkey option)[https://golang.org/pkg/text/template/#Template.Option] 
it can return only `"<no value>"`,`0` or raise error

_also can be seen [here](https://golang.org/src/text/template/exec_test.go?h=missingkey) in tests_

And that's not the only problem. As i tested it the option doesn't work at all in your case.
And there comes to play `go` versions unfortunately.

I found this [commit](https://github.com/golang/go/commit/277bcbbdcd26f2d64493e596238e34b47782f98e) 
which is very much related. It fixes former problem, that `missingkey=error` didn't raised error when i quote: `provided data is a nil-interface`.  If you look to the tags, the change was integrated only to `go` versions `1.8+`

This is exactly your case. You are passing `nil` value to the `Execute` method [here](https://github.com/seznam/goenvtemplator/blob/3496e445763eed716d2de7fa211c0bd98af72d01/template.go#L58)


As I`m running `1.7` version i tried this hack:
If you pass there empty structure and set the option to `missingkey=error` than you get error!
Hooray, but still this is not your intention but at least the option is working _somehow_.

After studying further the implementation.. I simply gave up :)
I don't know why the `zero` option doesn't work. And even if, it still wouldn't help you. You dont want a zero.

You would need to override template function which looks for the key, and instead of `<no value>` or anything else return the empty string.

But during studying of this issue a finally read through whole goenvtemplator and realized this testcase is completely out of goenvtemplator scope. You are testing acessing data passed to the mentioned `Execute` method but goenvtemplator *never* pases any values so IMHO there is no need to test this.
User simply can't get to this situation.

Other thing is to test 
```
# cat a 
K={{ env "NONEXISTING" }}
# ./goenvtemplator2 -template $PWD/a:/dev/stdout
K=
``` 

Which is ok. 

So my opinion after all this digging is that you should just change the testing templates to 
```
{`K={{ env "NONEXISTING" }}`, `K=`, nil},
{`K={{ env "NONEXISTING" | default "default value" }}`, `K=default value`, nil},
```

Hope it helps somehow :)

dohnto commented:
@FUSAKLA You are correct, the output I have provided is not correct (does not reflect my changes in code).

I have updated code so it works with older version of golang but it changes behaviour: it raises an error now if unknown ken is seen (but all of them are unknown). I have updated the testsuite. See my changed in 	b9c584d .

Also I would probably wait until @lksv comments why had a particular reason for switching to `html/template`.

dohnto commented:
Ok, so there is no particual reason for using `html/template` over `text/template`
